packages/core/src/analytics/Analytics.js

```diff
@@ -330,7 +330,12 @@ class Analytics {
     try {
       this.userInstance.anonymize();
 
-      this.handleOnUserChanged();
+      await this.handleOnUserChanged();
+
+      // Reset the user promise to force all tracks to wait for the next fetch of user data (guest user).
+      this.setUserPromise = new Promise(resolve => {
```

Comment:
Same as in the PR for next, this should come before the `handleOnUserChanged` call.